core/src/main/java/novoda/lib/sqliteprovider/util/UriUtils.java

```diff
@@ -6,29 +6,28 @@
 
 public class UriUtils {
 
-    private static Map<String, String> mappedIds = new HashMap<String, String>();
+    private UriUtils() throws IllegalAccessException {
+        throw new IllegalAccessException("Utils class shouldn't be instantiated");
+    }
 
-    public static UriUtils from(Uri uri) {
-        UriUtils utils = new UriUtils();
+    public static Map<String, String> mapIds(Uri uri) {
+        Map<String, String> mappedIds = new HashMap<>();
 
         final List<String> segs = uri.getPathSegments();
         String parent = "";
 
         for (int i = 0; i < segs.size(); i++) {
```

Comment:
Since this will potentially run on very small List of Strings, it will not be a problem but actually this `fori` would potentially be slower because it does `segs.size()` on each loop. 
I agree that it is out of scope.